src/NUnitFramework/framework/Constraints/EqualConstraint{T}.cs

```diff
@@ -0,0 +1,362 @@
+            return this;
+        }
+
+        /// <summary>
+        /// Flag the constraint to use the supplied Comparison object.
+        /// </summary>
+        /// <param name="comparer">The IComparer object to use.</param>
+        /// <returns>Self.</returns>
+        public EqualConstraint<T> Using(Comparison<T> comparer)
+        {
+            _comparer = (l, r) => comparer(l, r) == 0;
+            _fallbackComparer.ExternalComparers.Add(EqualityAdapter.For(comparer));
+            return this;
+        }
+
+        /// <summary>
+        /// Flag the constraint to use the supplied IEqualityComparer object.
+        /// </summary>
+        /// <param name="comparer">The IComparer object to use.</param>
+        /// <returns>Self.</returns>
+        public EqualConstraint<T> Using(IEqualityComparer comparer)
+        {
+            _comparer = (l, r) => comparer.Equals(l, r);
+            _fallbackComparer.ExternalComparers.Add(EqualityAdapter.For(comparer));
+            return this;
+        }
+
+        /// <summary>
+        /// Flag the constraint to use the supplied IEqualityComparer object.
+        /// </summary>
+        /// <param name="comparer">The IComparer object to use.</param>
+        /// <returns>Self.</returns>
+        public EqualConstraint<T> Using(IEqualityComparer<T> comparer)
+        {
+            _comparer = (l, r) => comparer.Equals(l, r);
+            _fallbackComparer.ExternalComparers.Add(EqualityAdapter.For(comparer));
+            return this;
+        }
+
+        #endregion
+
+        #region Public Methods
+
+        /// <summary>
+        /// Test whether the constraint is satisfied by a given value
+        /// </summary>
+        /// <param name="actual">The value to be tested</param>
+        /// <returns>True for success, false for failure</returns>
+        public override ConstraintResult ApplyTo<TActual>(TActual actual)
+        {
+            if (actual is T a)
+            {
+                bool isSuccess = false;
+
+                if (_comparer is null)
+                {
+                    GetTolerance(ref _tolerance);
+                    isSuccess = _fallbackComparer.AreEqual(_expected, a, ref _tolerance);
+                }
+                else
+                {
+                    isSuccess = _comparer(_expected, a);
+                }
+
+                return new EqualConstraintResult(this, GetConstraintResultData(), actual, isSuccess);
+            }
+            else
+            {
+                var value = _fallbackComparer.AreEqual(_expected, actual, ref _tolerance);
+                return new EqualConstraintResult(this, GetConstraintResultData(), actual, value);
+            }
+
+            throw new ArgumentException("Wrong type", nameof(actual));
+
+            EqualConstraintResult.ResultData GetConstraintResultData() => new()
+            {
+                // TODO: Handle string-specific cases separately?
```

Comment:
After some experimentation, I believe we'll have to add these here onto the generic constraint too. The below test will pass with NUnit 4.2.1 but will give a compiler error presently

```
[Test]
public void TestSomething()
{
    var a = (2, 5, "HELLO");
    var b = (2, 5, "hello");

    Assert.That(a, Is.EqualTo(b).IgnoreCase);
}
```